src/Models/Constants/Types.ts

```diff
@@ -188,9 +188,9 @@ export type IConsoleLogFunction = (
     ...args: unknown[]
 ) => void;
 
-export type MissingAzureRoleDefinitionAssignments = {
-    enforced?: Array<AzureAccessPermissionRoles>;
-    alternated?: Array<AzureAccessPermissionRoles>;
+export type AzureAccessPermissionRoleGroups = {
+    enforced: Array<AzureAccessPermissionRoles>; // roles must exist/required (and)
+    interchangeables: Array<Array<AzureAccessPermissionRoles>>; // within each group one or the other role has to exist (either/or)
```

Comment:
super nit: would make it singular `interchangable`

Reply:
yeah, I thought about that but decided to keep as plural to make it clear it is array of interchangeable arrays